// bbc-goodfood-scraped/index.js
var request = require('request');
var cheerio = require('cheerio');

module.exports = function(url) {       

    request(url, function(error, response, html){
        var $ = cheerio.load(html);
        var title, image, prep, cook, difficulty, serves;
        var ingredients = [];
        
        
        $('body').filter(function(){
            var data = $(this);
            title = $(".recipe-header__details-first > h1").text().trim();
            image = $(".img-container > img").attr('src');
            $( "ul.ingredients-list__group > li" ).each(function( index ) {
                $(this).find('.gf-tooltip').remove();
                $(this).find('span').remove();
                ingredients.push($( this ).text());  
            });
            prep = $(".recipe-details__cooking-time-prep").text().replace(/ Prep: /g,'').trim();
            cook = $(".recipe-details__cooking-time-cook").text().replace(/ Cook: /g,'').trim();
            difficulty = data.find(".recipe-details__item--skill-level").text().trim();
            serves = $(".recipe-details__item--servings > span").text().replace(/Serves/g,'').trim();
            
            data = {
                title: title,
                image: image,
                ingredients: ingredients,
                prep: prep,
                cook: cook,
                difficulty: difficulty,
                serves: serves,
                publisher: "BBC Good Food",
                recipe_url: url
            };
            return data;
        });
    });
    
};
